### PR TITLE
Add documentation for "search on references"

### PR DESCRIPTION
**Pull request by madtrick**
### Why this change

Because we are nice people who like their customers and want them to know how to use the product

### CODE REVIEW

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 1]
```diff
@@ -716,6 +716,75 @@ This example will return only image assets.
...
+- `content_type=article&fields.writtenBy.sys.contentType.sys.id=post&fields.writtenBy.fields.languages[all]=spanish,english`
+
+This a is comprehensive list of all the filters that can be used when searching on references: `all`, `in`, `nin`, `exists`, `match`, `gt`, `gte`,`lt`, `lte`, `ne`, `near` and `within`. The semantics of this filters is the same as when used on a non referenced entry.
+
+### Restrictions
+
+Searching on references has the following restrictions:
+
+- Searching on references can only be used on fields which link to a single entry. Fields which hold references to many entries or fields with references to assets are not supported by this functionality.
+- The maximum number of reference searches in one query is X. A bigger value will return an error.
```
Comment (ChrisChinchilla) [inline apiary/_partials/delivery-and-preview-resources.apib]: > X

What's 'x'?

**Comment by ChrisChinchilla** [comment]
@madtrick Pending an answer to that comment I have made an edit, let me know if I got anything wrong.

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 2]
```diff
@@ -716,6 +716,75 @@ This example will return only image assets.
 
   + Attributes (Empty Array)
 
+## Search on references [/spaces/{space_id}/assets?access_token={access_token}&content_type={content_type}&fields.{reference_field}.sys.contentType.sys.id={reference_content_type}&fields.{reference_field}.fields.{search_field}={search_value}]
+It's possible to search for entries based on the values of referenced entries. To understand this better lets firts consider the following scenario: entries of content type `Article` have a `writtenBy` field which is used to link to entries from content type `Author` (which among others have a `name` field). Using the search on references functionality we can, with just one API request, find those articles which have been writen by an author with a given name.
```
Comment (hlabas) [inline apiary/_partials/delivery-and-preview-resources.apib]: **It is** possible to search for entries based on the values of **their** referenced entries. To understand this better **let's first** consider the following scenario: entries of content type `Article` have a `writtenBy` field which **references entries of** content type `Author` (which **themselves** have a `name` field). **By using this functionality** we can, with just one API request, find ~~those~~ articles which have been **written** by an author with a given name.

**Comment by madtrick** [comment]
@ChrisChinchilla thx for adding the tags. Completely forgot about it

File: apiary/_partials/delivery-and-preview-resources.apib (hunk 2 above)
Comment (ChrisChinchilla) [inline apiary/_partials/delivery-and-preview-resources.apib]: @hlabas 

This was on an outdated version, could you take another look?

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 3]
```diff
@@ -716,6 +716,90 @@ This example will return only image assets.
...
+    + access_token (required, string, `b4c0n73n7fu1`) ... :[token description](tokentype)
+    + content_type (required, string, `cat`) ...
+    + reference_content_type (required, string, `cat`) ... The id of then content type of the referenced entry
+    + reference_field (required, string, `bestFriend`) ... The id of the field holding the reference
+    + search_field (required, string, `color`) ... The id of the field you want to search
+    + search_value (required, string, `rainbow`) ... The value you are looking for
+
+### Search on references [GET]
+
+This example will return cats who are friends with another cat of 'rainbow' color.
```
Comment (hlabas) [inline apiary/_partials/delivery-and-preview-resources.apib]: Not sure I get what's in here

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 4]
```diff
@@ -716,6 +716,90 @@ This example will return only image assets.
...
+
+### Structure of a query
+
+Here's how the example above would look as a query:
+
+```sql
+content_type=article&fields.writtenBy.sys.contentType.sys.id=author&fields.writtenBy.fields.name=john
+```
+
+- First is the `content_type` parameter which you must include any time you want to filter based on the values of the fields of entries of a certain content type.
```
Comment (ChrisChinchilla) [inline apiary/_partials/delivery-and-preview-resources.apib]: @hlabas Sorry, I don't really understand what you're saying… yeah, there's a lot of 'of's, but I think it's OK.

File: apiary/_partials/delivery-and-preview-resources.apib (hunk 3 above)
Comment (ChrisChinchilla) [inline apiary/_partials/delivery-and-preview-resources.apib]: @hlabas Made small tweak, but I will be overhauling all examples in our docs soon anyway :)

File: apiary/_partials/delivery-and-preview-resources.apib (hunk 4 above)
Comment (hlabas) [inline apiary/_partials/delivery-and-preview-resources.apib]: Alright then. Read strange to me, but I don't mind.

File: apiary/_partials/delivery-and-preview-resources.apib (hunk 1 above)
Comment (madtrick) [inline apiary/_partials/delivery-and-preview-resources.apib]: @ChrisChinchilla done.

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 5]
```diff
@@ -716,6 +716,90 @@ This example will return only image assets.
...
+### Structure of a query
+
+Here's how the example above would look as a query:
+
+```sql
+content_type=article&fields.writtenBy.sys.contentType.sys.id=author&fields.writtenBy.fields.name=john
+```
+
+- First is the `content_type` parameter which you must include when you want to filter based on the values of the fields of entries of a certain content type.
+- Second is `fields.writtenBy.sys.contentType.sys.id=post` which you use to to filter on fields of entries from content type `Author`.
```
Comment (TimBeyer) [inline apiary/_partials/delivery-and-preview-resources.apib]: Shouldn't this be `post`?

Comment (madtrick) [inline apiary/_partials/delivery-and-preview-resources.apib]: Giphy

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 6]
```diff
@@ -716,6 +716,90 @@ This example will return only image assets.
...
+```
+
+
+### Available nested filters
+You can use the `all`, `in`, `nin`, `exists`, `match`, `gt`, `gte`,`lt`, `lte`, `ne`, `near` and `within` filters when searching on references.
+
+For example, all the following queries are valid:
+
+```sql
+content_type=article&fields.writtenBy.sys.contentType.sys.id=post&fields.writtenBy.fields.age[gt]=21
```
Comment (madtrick) [inline apiary/_partials/delivery-and-preview-resources.apib]: Giphy

File: apiary/_partials/delivery-and-preview-resources.apib (hunk 5 above)
Comment (madtrick) [inline apiary/_partials/delivery-and-preview-resources.apib]: Fixed

File: apiary/_partials/delivery-and-preview-resources.apib (hunk 6 above)
Comment (madtrick) [inline apiary/_partials/delivery-and-preview-resources.apib]: Fixed

**Pull request merged**